enterprise/server/auth/auth.go

```diff
@@ -20,10 +23,35 @@ var (
 )
 
 func Register(ctx context.Context, env environment.Env) error {
-	authenticator, err := oidc.NewOpenIDAuthenticator(ctx, env, *adminGroupID)
+	httpAuthenticators := []interfaces.HTTPAuthenticator{}
+	userAuthenticators := []interfaces.UserAuthenticator{}
+
+	oidc, err := oidc.NewOpenIDAuthenticator(ctx, env, *adminGroupID)
 	if err != nil {
-		return status.InternalErrorf("Authenticator failed to configure: %v", err)
+		return status.InternalErrorf("OIDC authenticator failed to configure: %v", err)
+	}
+	httpAuthenticators = append(httpAuthenticators, oidc)
+	userAuthenticators = append(userAuthenticators, oidc)
+
+	if saml.SamlEnabled(env) {
+		samlAuthenticator := saml.NewSAMLAuthenticator(env)
+		httpAuthenticators = append(httpAuthenticators, samlAuthenticator)
```

Comment:
do you want the SAML authenticator to come first in these lists so that it takes precedence?

Reply:
Ideally order shouldn't matter, and if for some reason it does I'd like to fix it

Reply:
ah I was mostly concerned about which error gets returned (since some codepaths depend on error codes), but after re-reading I see we're returning `errors[0]` which effectively prioritizes OIDC and keeps the same behavior we have today